src/auth/auth.controller.ts

```diff
@@ -14,8 +15,14 @@ export class AuthController {
   constructor(
     private readonly authService: AuthService,
     private readonly passwordService: PasswordService,
+    private readonly walletService: WalletService,
   ) {}
 
+  @Get('validate/name/:name')
+  async validateName(@Param('name') name: string): Promise<boolean> {
+    return await this.walletService.validateName(name);
```

Comment:
Lets not add DI on `walletService` unnecessarily here and move the `validateName` function inside authService so we can use `return await this.authService.validateWalletName(name)` or authService.validateName(name)